lib/game_client/ogs/ogs_websocket_manager.dart

```diff
@@ -32,7 +34,10 @@ class OGSWebSocketManager {
   int _lastRequestId = 0;
   final Map<int, Completer<dynamic>> _pendingRequests = {};
 
-  OGSWebSocketManager({required this.serverUrl}) {
+  OGSWebSocketManager({
+    required this.serverUrl,
+    StreamChannel Function(Uri)? createChannel,
```

Comment:
I guess this can be just a public field and use a default value? You cannot mutate it anyway.
```dart
OGSWebSocketManager({
    required this.serverUrl,
    this.createChannel = WebSocketChannel.connect,
...
```